src/main/app/claims/claimStoreClient.ts

```diff
@@ -15,21 +16,35 @@ export const claimStoreApiUrl: string = `${claimApiBaseUrl}/claims`
 const claimStoreResponsesApiUrl: string = `${claimApiBaseUrl}/responses/claim`
 
 export class ClaimStoreClient {
-  static saveClaimForUser (draft: Draft<DraftClaim>, user: User): Promise<Claim> {
+  constructor (private request: RequestPromiseAPI = requestPromiseApi) {
+    // Nothing to do
+  }
+
+  saveClaim (draft: Draft<DraftClaim>, claimant: User): Promise<Claim> {
     const convertedDraftClaim = ClaimModelConverter.convert(draft.document)
-    return request
-      .post(`${claimStoreApiUrl}/${user.id}`, {
+    return this.request
+      .post(`${claimStoreApiUrl}/${claimant.id}`, {
         body: convertedDraftClaim,
         headers: {
-          Authorization: `Bearer ${user.bearerToken}`
+          Authorization: `Bearer ${claimant.bearerToken}`
+        }
+      })
+      .then(claim => {
+        return new Claim().deserialize(claim)
+      })
+      .catch((err) => {
+        if (err.statusCode === HttpStatus.CONFLICT) {
+          return this.retrieveByExternalId(draft.document.externalId, claimant)
```

Comment:
I would love to have some logs here stating that conflict occurred and we retrieved claim from data store.